db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM.class/__methods__/update_provision_status.rb

```diff
@@ -1,16 +1,21 @@
 #
-# Description: Update provision status.
+# Description: This method updates the provision status.
 # Required inputs: status
 #
 
-prov   = $evm.root['miq_provision']
+prov = $evm.root['miq_provision']
 unless prov
   $evm.log(:error, "miq_provision object not provided")
   exit(MIQ_STOP)
 end
 status = $evm.inputs['status']
 
-# Update Status for on_entry,on_exit
-if $evm.root['ae_result'] == 'ok' || $evm.root['ae_result'] == 'error'
-  prov.message = status
-end
+# Update Status Message
+updated_message  = "[#{$evm.root['miq_server'].name}] "
+updated_message += "VM [#{prov.get_option(:vm_target_name)}] "
+updated_message += "Step [#{$evm.root['ae_state']}] "
+updated_message += "Status [#{status}] "
```

Comment:
@mkanoor @tinaafitz 
Not sure, I think if any of these fields are nil we have bigger problems.  
Which fields are you worried about ?